docs/source/getting-started.rst

```diff
@@ -0,0 +1,306 @@
+
+Then to run the container in dev mode just use the following command::
+
+    $ surround run dev
+
+This will run the container linking the folder ``testproject/testproject`` with the working directory in the
+container. So during development when you make small changes, there is no need to build the image, just run
+this command again.
+
+Then when you are ready for production you can use the following command::
+
+    $ surround run prod
+
+Which will first build the image and then run the container without any linking to the host machine.
+The image created in the build can also then be committed to a Docker Hub repository and shared.
+
+.. note:: Both ``dev`` and ``prod`` will use the command ``$ python3 -m testproject`` inside the container.
+        Meaning they will use the default mode set in ``__main__.py`` (which is ``batch`` in default projects) 
+        when running the pipeline.
+
+The following commands will force which mode to use::
+
+    $ surround run batch
+    $ surround run train
+
+Serving your first pipeline via Web Endpoint
+********************************************
+
+When generating a project, you get asked::
+    
+    Does it require a web runner? (y/n)
+    
+If we say yes to this then Surround will generate a generic ``batch_runner.py`` but it will also
+generate a new script called ``web_runner.py``. 
+
+This script contains a new ``Runner`` which will use `Tornado <https://www.tornadoweb.org/en/stable/>`_
+to host a web server which will allow your pipeline to be accessible via HTTP request. By default the 
+``WebRunner`` will host two endpoints:
+
+- ``/info`` - access via GET request, will return ``{'version': '0.0.1'}``
+- ``/estimate`` - access via POST request, body must have a JSON document containing input data::
+
+    {
+        "message": "this text will be processed" 
+    }
+
+So lets create a new pipeline that does the same data processing as the one in :ref:`create-first-pipeline` but
+we will send strings via web endpoint and get the results in the response of the request.
+
+First generate a new project, this time saying yes to the require web prompt, and make all the changes we did in
+:ref:`create-first-pipeline` and test it is still working locally.
+
+Next we are going to build an image for our pipeline using the command::
+
+    $ surround run build
+
+Then we are going to run our default server using the command::
+
+    $ surround run web
+
+You should get output like so::
+
+    INFO:root:Server started at http://localhost:8080
+
+.. note:: If you would like to run it on the host machine instead of in a container, you must install Tornado using
+        this command: ``$ pip3 install tornado==6.0.2``
+
+Now hopefully if you load ``http://localhost:8080/info`` in your preferred browser, you should see the following::
+
+    {"version": "0.0.1"}
+
+.. note:: If you are running this on Windows and don't see the above, try using ``http://192.168.99.100:8080/info`` instead.
```

Comment:
Is that the default ip address registered by Tornado in Windows?

Reply:
Its the default IP address used for Docker containers on Windows, so if it is ran locally it will be localhost, but if ran in Docker, they will need to use that IP

Reply:
Cool :+1:
That should be fine then